### PR TITLE
ci: update PR-commenter and Mergify for Kubernertes 1.26/1.30

### PR DESCRIPTION
**Pull request by nixpanic**
The k8s-external-storage/1.26 CI job does not work with the versions
that are part of the release-v3.11 and devel branches.

Run CI jobs with Kubernetes 1.30 for the devel branch and upcoming
releases.

Related-to: #4569

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ environments.
 
 | Ceph CSI Version | Container Orchestrator Name | Version Tested     |
 | -----------------| --------------------------- | -------------------|
+| v3.11.1          | Kubernetes                  | v1.27, v1.28, v1.29, v1.30|
 | v3.11.0          | Kubernetes                  | v1.26, v1.27, v1.28, v1.29|
```
Comment (Rakshith-R) [inline README.md]: We don't have 3.11.1 yet. Let's add the version when we do the release.

Comment (nixpanic) [reply README.md]: I added that so we won't forget to update it. Do you prefer a note in `PendingReleaseNotes.md` instead?

Comment (Rakshith-R) [reply README.md]: We can just check while doing the release and update in Readme IMO

Comment (nixpanic) [reply README.md]: Ok, removed the README.md change.

File: .mergify.yml [hunk 2]
```diff
@@ -34,18 +34,18 @@ queue_rules:
...
               - "status-success=ci/centos/mini-e2e-helm/k8s-1.28"
               - "status-success=ci/centos/mini-e2e-helm/k8s-1.29"
-              - "status-success=ci/centos/mini-e2e/k8s-1.26"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.30"
               - "status-success=ci/centos/mini-e2e/k8s-1.27"
               - "status-success=ci/centos/mini-e2e/k8s-1.28"
               - "status-success=ci/centos/mini-e2e/k8s-1.29"
+              - "status-success=ci/centos/mini-e2e/k8s-1.30"
               - "status-success=ci/centos/upgrade-tests-cephfs"
               - "status-success=ci/centos/upgrade-tests-rbd"
```
Comment (Rakshith-R) [inline .mergify.yml:50]: k8s 1.30 needs to be added for release-v3.11 below too ?

**Review by Rakshith-R — APPROVED** [review]
Thanks

**Review by iPraveenParihar — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**